### DISCUSSION TITLE
Proposition for fixing the error with multiple re-renders

### DISCUSSION THREAD

**Discussion by alefDev-prog** [ekqt/screenshot#31] [Ideas]
Hi,

I was very curious on how one may fix the error with the `ClipboardImage` component re-rendering several times on one click. I understand that you may have your own idea on how to deal with this but I just wanted to share how I have dealt with this problem.

What I did:

1. Used the `userRef` hook to reference the image element
2. Included a `handleClick` function and within that function I call the `pasteImage`
3. Used the `useEffect` hook and transitioned the "load" eventListener to it.

These changes resulted in that the `pasteImage` function only get called once per click.
I will provide a link to the code if you want to have a look at it: [code](https://github.com/alefDev-prog/screenshot/blob/testBranch/testClipboardImage.tsx)

**Comment by iamhectorsosa** [comment]
Let's try to address this with your open PR. I will draw inspo from your suggestion and work on styling those Toasts as well! Huge thanks, @alefDev-prog

**Comment by iamhectorsosa** [comment]
@alefDev-prog Updating Handlers in Clipboard Image component #51

**Reply by alefDev-prog** [reply]
This look very good! I'm happy I could provide some inspiration for the solution.
